dms/models/dms_security_mixin.py

```diff
@@ -289,6 +304,32 @@ def filter_access_groups(self, operation):
             ids_with_access += list(map(lambda val: val[0], self.env.cr.fetchall()))
         return self & self.browse(ids_with_access)
 
+    def _get_directory_ids_with_res_model_without_access(self, operation):
+        """
+        It's necessary to get all directories with res_model related and check
+        if access to related record.
+        Context use_res_model_without_access=False is used to skip some
+        part of functions that use these function and prevent recursion error.
+        """
+        return (
+            self.env["dms.directory"]
+            .with_context(use_res_model_without_access=False)
+            .search(
+                [
+                    ("res_model", "!=", False),
+                    "|",
+                    "&",
+                    ("is_root_directory", "=", True),
+                    ("root_storage_id.inherit_access_from_parent_record", "=", True),
+                    "&",
+                    ("is_root_directory", "=", False),
+                    ("storage_id.inherit_access_from_parent_record", "=", True),
+                ]
+            )
+            .filtered(lambda x: not x.check_access("read"))
+            .ids
```

Comment:
Why returning `ids`, which always gives you less options in case you need any other file? Unless the method is orm_cached, return always recordsets